supervisor.go

```diff
@@ -40,8 +42,11 @@ func (*nopLocker) Unlock() {}
 type Supervisor interface {
 	io.Closer
 
+	// WithContext sets the context.
+	WithContext(ctx context.Context)
```

Comment:
Since you removed the argument name from the WithPumps method, I would remove `ctx` name as well ;)